testlib/s3mock.py

```diff
@@ -202,58 +166,16 @@ def ln(part):
     return b"\n".join(output)
 
 
-class RecordingAdapterShim:
```

Comment:
I sort of used the recording feature, but not much. I decided to nix it and replace it with some code that shows what should have happened at the THWARTING point. It's not quite the same, but I think it still solves all my use cases.